src/raft_server.c

```diff
@@ -1084,6 +1097,7 @@ static raft_node_t* raft_add_non_voting_node_internal(raft_server_t* me_, raft_e
     if (!node)
         return NULL;
 
+    raft_node_set_addition_committed(node, 0);
```

Comment:
Looks like it would be more intuitive to keep raft_node_new unchanged...

Reply:
There's a difference: 

raft_add_[non_voting]_node() is called in response to a log entry. We don't want the addition to be "finalized" until that log entry has been applied.

But raft_node_new() is a private call -- for example to create an initial list of replicas.